### PR TITLE
Lint

### PR DESCRIPTION
**Pull request by dkfellows**
These are the linting changes for sPyNNaker. They depend on some changes elsewhere, so this build is currently broken.

### CODE REVIEW

File: spynnaker/pyNN/models/neuron/synapse_io/synapse_io_row_based.py [hunk 1]
```diff
@@ -229,7 +229,7 @@ def get_synapses(
         max_delayed_row_length = 0
         stages = numpy.zeros(0, dtype="uint32")
         delayed_source_ids = numpy.zeros(0, dtype="uint32")
-        if len(delayed_connections) > 0:
+        if delayed_connections:
```
Comment (rowleya) [inline spynnaker/pyNN/models/neuron/synapse_io/synapse_io_row_based.py]: delayed_connections is a numpy array

File: spynnaker/pyNN/overridden_pacman_functions/graph_edge_filter.py [hunk 2]
```diff
@@ -27,53 +27,58 @@ def __call__(self, machine_graph, graph_mapper):
...
-                    new_machine_graph_partition.add_constraints(
-                        partition.constraints)
-                else:
-                    logger.debug("this edge was pruned {}".format(edge))
-            progress_bar.update()
-        progress_bar.end()
+            for edge in progress.over(partition.edges, False):
+                if self._is_filterable(edge, graph_mapper):
+                    logger.debug("this edge was pruned %s", edge)
+                    continue
```
Comment (rowleya) [inline spynnaker/pyNN/overridden_pacman_functions/graph_edge_filter.py]: Is continue really better than else here?

File: spynnaker/plot_utils.py [hunk 3]
```diff
@@ -10,7 +11,7 @@
 
 
 def _precheck(data, title):
-    if data is None or len(data) == 0:
+    if not data:
```
[the remark this answers is not in the document]
Comment (dkfellows) [reply spynnaker/plot_utils.py]: OK

File: spynnaker/pyNN/models/common/spike_recorder.py [hunk 4]
```diff
@@ -43,59 +44,57 @@ def get_n_cpu_cycles(self, n_neurons):
     def get_spikes(
             self, label, buffer_manager, region, placements, graph_mapper,
             application_vertex, machine_time_step):
-
+        # pylint: disable=too-many-arguments, too-many-locals
         spike_times = list()
         spike_ids = list()
-        ms_per_tick = machine_time_step / 1000.0
+        ms_per_tick = float(machine_time_step / 1000.0)
```
[the remark this answers is not in the document]
Comment (dkfellows) [reply spynnaker/pyNN/models/common/spike_recorder.py]: Yes

File: spynnaker/pyNN/abstract_spinnaker_common.py [hunk 5]
```diff
@@ -379,10 +379,9 @@ def get_projections_data(self, projection_to_attribute_map):
                     "MemoryExtraMonitorToChipMapping"))
 
         # set up the router timeouts to stop packet loss
-        if using_extra_monitor_functionality:
-            for data_receiver, extra_monitor_cores in receivers:
-                data_receiver.set_cores_for_data_extraction(
-                    self._txrx, list(extra_monitor_cores), self._placements)
+        for data_receiver, extra_monitor_cores in receivers:
```
[the remark this answers is not in the document]
Comment (dkfellows) [reply spynnaker/pyNN/abstract_spinnaker_common.py]: The checking of the flag is guarding the filling out of the `receivers` list; when the `using_extra_monitor_functionality` flag isn't set, that list is consistently empty.

File: spynnaker/pyNN/abstract_spinnaker_common.py [hunk 6]
```diff
@@ -394,24 +393,24 @@ def get_projections_data(self, projection_to_attribute_map):
                 mother_lode.set(projection, attribute, data)
 
         # reset time outs for the receivers
-        if using_extra_monitor_functionality:
-            for data_receiver, extra_monitor_cores in receivers:
-                data_receiver.unset_cores_for_data_extraction(
-                    self._txrx, list(extra_monitor_cores), self._placements)
+        for data_receiver, extra_monitor_cores in receivers:
```
[the remark this answers is not in the document]
Comment (dkfellows) [reply spynnaker/pyNN/abstract_spinnaker_common.py]: As above re response

File: spynnaker/pyNN/models/neural_projections/connectors/abstract_connector.py [hunk 7]
```diff
@@ -104,20 +91,19 @@ def set_projection_information(
...
         if (not numpy.isscalar(values) and
-                not (globals_variables.get_simulator().
-                     is_a_pynn_random(values)) and
+                not (get_simulator().is_a_pynn_random(values)) and
                 not hasattr(values, "__getitem__")):
             raise Exception("Parameter {} format unsupported".format(name))
         if not allow_lists and hasattr(values, "__getitem__"):
             raise NotImplementedError(
-                "Lists of {} are not supported the implementation of"
+                "Lists are not supported the implementation of"
```
[the remark this answers is not in the document]
Comment (dkfellows) [reply spynnaker/pyNN/models/neural_projections/connectors/abstract_connector.py]: OK, seems fair enough

File: spynnaker/pyNN/abstract_spinnaker_common.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (rowleya) [reply spynnaker/pyNN/abstract_spinnaker_common.py]: Oh yes - I missed this as was hidden by github!

**Pull request merged**